### ISSUE TITLE
Doesn't work with different network

### ISSUE THREAD

**Issue by s21v1d9p** [sowmen/Android-P2P-Chat-Messenger-using-Java-TCP-IP-Socket-Programming#1]
The following application is working only on same network .
Can we implement working on different network for
Different ISP.

**Comment by sowmen** [comment]
> The following application is working only on same network .
> Can we implement working on different network for
> Different ISP.

The client tries to make a connection to an IP and port. It doesn't care if it's on the same network or out somewhere on the internet. You have to configure your router or network to enable port forwarding. By default routers block socket addresses that are not listed on their NAT tables. 

[https://coderanch.com/t/667020/java/Socket-connections-networks](url)
[https://stackoverflow.com/questions/51323963/java-socket-client-to-server-on-different-network](url)

**Comment by s21v1d9p** [comment]
Did you run this using mobile network ?
You can then understand what i am trying to say you

**Comment by sowmen** [comment]
I did not test it with port forwarding. I tested it
1. With both devices connected to the same wifi
2. One device connected to the hotspot of another device.

To communicate between two devices connected to different networks either Data connection or different wifi networks the ip addresses need to be globally routable ip addresses.

With wifi, the router blocks any local or non routable addresses. This can be overcome using port forwarding.

With data connection this is only possible if your carrier gives out static public ip addresses. So, this cannot be done as of now. One way to overcome this is by using a dummy server that can be accessed by both devices and forward data via that server.

[https://stackoverflow.com/questions/811298/is-it-possible-to-connect-a-socket-between-two-mobile-devices-iphones-over-the](url)
[https://serverfault.com/questions/655927/possible-for-socket-communication-between-2-internet-connected-gsm-modems](url)
